tasks/tasks.go

```diff
@@ -55,28 +51,7 @@ func Sync(ctx context.Context, config *tsbridge.Config) error {
 		return err
 	}
 
-	var errSync []error
-
-	sdClientOnce.Do(func() {
-		sdClient, err = stackdriver.NewAdapter(ctx, config.Options.SDLookBackInterval)
-		if err != nil {
-			errSync = append(errSync, fmt.Errorf("unable to initialize stackdriver adapter: %v", err))
-		}
-	})
-
-	statsCollectorOnce.Do(func() {
-		statsCollector, err = tsbridge.NewCollector(ctx, config.Options.SDInternalMetricsProject, config.Options.MonitoringBackends)
-		if err != nil {
-			errSync = append(errSync, fmt.Errorf("unable to initialize stats collector: %v", err))
-		}
-	})
-
-	// Process errors from Once.Do blocks since we cannot return in those
-	if len(errSync) > 0 {
-		return fmt.Errorf("errors occured during sync init: %v", errSync)
-	}
-
-	if errs := tsbridge.UpdateAllMetrics(ctx, metrics, sdClient, config.Options.UpdateParallelism, statsCollector); errs != nil {
+	if errs := tsbridge.UpdateAllMetrics(ctx, metrics, config.Dependencies.SDClient, config.Options.UpdateParallelism, config.Dependencies.StatsCollector); errs != nil {
```

Comment:
Sync is called in a loop, and each loop we're passing the connection handlers back and forth as args.

It might be cleaner to have a struct that holds these handlers, so that in the future if we add more external handlers, they can just be added to the struct and not passed around as args for every loop.
 
Maybe wrap UpdateAllMetrics in a struct?

```
type UpdateMetrics struct {
 sdClient       *stackdriver.Adapter
 statsCollector *tsbridge.StatsCollector
}

func New(ctx,  *stackdriver.Adapter, *tsbridge.StatsCollector) *UpdateMetrics {
 return &UpdateMetrics {
    ...
 } 
}

func (u *UpdateMetrics) All(ctx, config.Options.UpdateParallelism) error {
 ...
}

```

WDYT?


(or even wrap it around config.ExternalDependencies)

Reply:
I wrapped around UpdateAllMetrics, wdyt?